### PR TITLE
11341 add dblclick event

### PR DESCRIPTION
**Pull request by elenatorro**
Related issue: https://github.com/CartoDB/cartodb/issues/11341

This PR  adds a double click listener that removes the anchor point that has been clicked by the user. I am not sure if this is the best way to solve this, opinions needed :)

polygon

### CODE REVIEW

File: src/api/v4/native/trigger-layer-feature-event.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ module.exports = function (eventName, internalEvent, layers) {
      * @type {carto.layer.FeatureEvent}
      * @api
      */
+    /**
```
Comment (IagoLast) [inline src/api/v4/native/trigger-layer-feature-event.js]: !

File: src/geo/cartodb-layer-group-view-base.js [hunk 2]
```diff
@@ -51,20 +51,20 @@ CartoDBLayerGroupViewBase.prototype = {
       this.interaction[layerIndexInLayerGroup] = new this.interactionClass()
         .map(this.nativeMap)
         .tilejson(tilejson)
-        .on('on', function (o) {
+        .on('on', function (event) {
```
Comment (IagoLast) [inline src/geo/cartodb-layer-group-view-base.js]: To be more consistent I think this also could be renamed to `zeraEvent`

**Comment by alonsogarciapablo** [comment]
Please make sure this works with Google Maps too! Thanks!

**Comment by IagoLast** [comment]
> Please make sure this works with Google Maps too! Thanks!

Just checked, works like a charm ;)

**Comment by elenatorro** [comment]
I added a couple of tests with two main use cases. I have been talking with @IagoLast and @ivanmalagon about the examples we are providing.

This feature works in the examples when editing an existing polygon, but if you create a new one you would need to bind manually the `dblclick` event. I think that for solving this issue, these changes are enough. However, I would like to discuss with you if you agree or if we should make more changes :)

File: test/spec/geo/geometry-views/shared-tests-for-path-views.js [hunk 3]
```diff
@@ -98,6 +98,31 @@ module.exports = function (Path, MapView, PathView) {
...
+            40
+          ]
+        });
+
+        this.geometry.setCoordinates([
+          [ -10, 10 ], [ 10, 20 ], [ 30, 40 ]
+        ]);
+
+        this.geometry.addPoint(pointToDelete);
+
```
Comment (IagoLast) [inline test/spec/geo/geometry-views/shared-tests-for-path-views.js]: Just a minor thing:

I would add a pre-spec to clarify the test

```javascript
expect(this.geometry.getCoordinates()).toEqual([[ -10, 10 ], [ 10, 20 ], [ 30, 40 ], [-40, 40]); // <---
pointToDelete.trigger('dblclick');
expect(this.geometry.getCoordinates()).toEqual([[ -10, 10 ], [ 10, 20 ], [ 30, 40 ]);
```

Comment (alonsogarciapablo) [reply test/spec/geo/geometry-views/shared-tests-for-path-views.js]: 🙇

Comment (elenatorro) [reply test/spec/geo/geometry-views/shared-tests-for-path-views.js]: 👍

**Pull request merged**